doc/release/upcoming_changes/25495.deprecation.rst

```diff
@@ -0,0 +1,9 @@
+`numpy.fft` deprecations for n-D transforms with ``None`` values in arguments
+-----------------------------------------------------------------------------
+
+`numpy.fft.fftn`, `numpy.fft.ifftn`, `numpy.fft.rfftn` and `numpy.fft.irfftn`
+are deprecated in line with the following part of the array API
+specification: "If ``s`` is not ``None``, ``axes`` must not be ``None``".
```

Comment:
Suggested wording change to the above:

```rst
Using `numpy.fft.fftn`, `numpy.fft.ifftn`, `numpy.fft.rfftn` and `numpy.fft.irfftn` with the ``s`` parameter set to value that is not `None` and the ``axes`` set to ``None`` has been deprecated, to match the array API standard.
```

Right now a reader might mistakenly read that the whole function is deprecated, not just this one usage.


If a user has `s` set to `None` or passes an array containing `None`, what should they do now to get the same behavior back? If that's still possible, you should explain how here as well as in the deprecation message if that would fit without being overly wordy there.